doc/releases/changelog-dev.md

```diff
@@ -111,6 +111,9 @@
 
 <h3>Bug fixes</h3>
 
+* Fixed a bug where caching together with JIT compilation and broadcasted tapes yielded wrong results
```

Comment:
We should probably also say what actually got changed.
```suggestion
* Fixed a bug where caching together with JIT compilation and broadcasted tapes yielded wrong results
  `Operator.hash` now depends on the memory location, `id`, of a Jax tracer instead of its string representation.
```